pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -602,6 +602,16 @@ public void testTimeFunc(boolean useMultiStageQueryEngine)
     assertEquals(oneHourAgoTodayStr, expectedOneHourAgoTodayStr);
   }
 
+  @Test
+  public void testMaxSerializedResponseLengthPerServer()
+      throws Exception {
+    String queryWithOption = "SET maxSerializedResponseLengthPerServer=1000; " + SELECT_STAR_QUERY;
+    JsonNode response = postQuery(queryWithOption);
+    assert response.get("exceptions").size() > 0;
+    String exceptionStr = response.get("exceptions").get(0).get("message").toString();
+    assert exceptionStr.contains("Serialized response exceeds threshold");
```

Comment:
I don't think it is a good idea to assert on messages. We should instead create a specific QueryException with its own error code and assert that the exception error code is that one.